src/i_joystick.c

```diff
@@ -294,14 +309,31 @@ static int GetAxisStateGamepad(int axis, int invert, int dead_zone)
 
     if (result < dead_zone && result > -dead_zone)
     {
-        result = 0;
+        return 0;
     }
 
     if (invert)
     {
         result = -result;
     }
 
+    result *= FRACUNIT / 32768; // Want FXP number between -1 and 1
+
+    switch (response_curve)
+    {
+        case STICK_RESPONSE_CUBIC:
+            // Allows for finer control when stick deflection is small.
+            result = FixedMul(FixedMul(result, result), result);
+            break;
+
+        case STICK_RESPONSE_LINEAR:
+        default:
```

Comment:
Having a `default` case defeats the whole point of having the compiler check it since the warning will never fire.